lib/connection.js

```diff
@@ -1137,28 +1160,145 @@ Connection.prototype.search = function(sosl, callback) {
+ * @param {Boolean} options.autofetch - whether to automatically fetch metadata for large numbers of 
+ *                         types (one batch request returns a maximum of 25 results); when true, will make 
+ *                         subsequent requests until all object metadata is fetched; when false (default), 
+ *                         will make one batch request for maximum of 25 results
+ * @param {Callback.<DescribeSObjectResult[]>} [callback] - Callback function
+ * @returns {Promise.<DescribeSObjectResult[]>}
+ */
+/**
+ * Batch describe SObject metadata
+ *
+ * @method Connection#batchDescribe
+ * @param {BatchDescribeSObjectOptions} options - options for function
+ * @param {String[]} options.types - names of objects to fetch
+ * @param {Boolean} options.autofetch - whether to automatically fetch metadata for large numbers of 
+ *                         types (one batch request returns a maximum of 25 results); when true, will make 
+ *                         subsequent requests until all object metadata is fetched; when false (default), 
+ *                         will make one batch request for maximum of 25 results
+ * @param {Callback.<DescribeSObjectResult[]>} [callback] - Callback function
+ * @returns {Promise.<DescribeSObjectResult[]>}
+ */
+Connection.prototype.batchDescribe = Connection.prototype.batchDescribeSObjects = function (
+  options,
+  callback
+) {
+  this.doBatchDescribeRequests(options)
+    .thenCall(callback);
+};
+
+/** private */
+Connection.prototype.doBatchDescribeRequests = function (
+  options
+) {
+  var sobjects = [];
+  var url = [this._baseUrl(), "composite/batch"].join("/");
+  var types = options.types;
+  var autofetch = options.autofetch || false;
+  var typesToFetch = types.length > MAX_BATCH_REQUESTS ? types.slice(0, MAX_BATCH_REQUESTS) : types;
+  var typesRemaining = types.length > MAX_BATCH_REQUESTS ? types.slice(MAX_BATCH_REQUESTS) : [];
+  var batchRequests = [];
+  typesToFetch.forEach(type => {
+    batchRequests.push({
+      method: "GET",
+      url: ["v" + this.version, "sobjects", type, "describe"].join("/")
+    });
+  });
+  return this.request({
+    method: "POST",
+    url,
+    body: JSON.stringify({ batchRequests }),
+    headers: {
+      "Content-Type": "application/json"
+    }
+  }).then((response) => {
+      if (response.results) {
+        var i = 0;
+        for (var subResp of response.results) {
+          if (Array.isArray(subResp.result)) {
+            if (subResp.result[0].errorCode && subResp.result[0].message) {
+              this._logger.error(
+                `Error: ${subResp.result[0].errorCode} ${subResp.result[0].message} - ${typesToFetch[i]}`
+              );
+            }
+          } else {
+            sobjects.push(subResp.result);
+          }
+          i++;
+        }
+      }
+      if (typesRemaining.length > 0 && autofetch) {
+        return this.doBatchDescribeRequests({ types: typesRemaining, autofetch }).then(
```

Comment:
We could still look into taking advantage of parallelization here.

Reply:
Ref: W-8035201

Reply:
Parallelized with `feat(conn): parallelize batch request` commit.